framework/auth/views.py

```diff
@@ -251,6 +288,68 @@ def confirm_email_get(token, auth=None, **kwargs):
     ))
 
 
+@collect_auth
+def unconfirmed_email_remove(auth=None):
+    """Called at login if user cancels their merge or email add.
+    methods: DELETE
+    """
+    user = auth.user
+    if user is None:
+        raise HTTPError(http.NOT_FOUND)
```

Comment:
404 isn't appropriate here; a 401 is used for when auth creds aren't passed. Why not use `must_be_logged_in`?